# Top Earners in Movie Industry

## Table of Contents

<ul>
    <li><a href="#intro">Introduction</a></li>
    <li><a href="#eda">Exploratory Data Analysis</a></li>
    <li><a href="#conclusion">Conclusion</a></li>
</ul>

<a id="#intro"></a>
## Introduction

> This analysis project is to be done using the imdb movie data. When the analysis is completed, you should be able to find the top 5 highest grossing directors, the top 5 highest grossing movie genres of all time, comparing the revenue of the highest grossing movies and which companies released the most movies. 

> There are 10 columns that will not be needed for the analysis. Use pandas to drop these columns. HINT: Only the columns pertaining to revenue will be needed.

> To get you started, I've already placed the needed code for getting the packages and datafile that you will be using for the project. 

In [1]:
import pandas as pd
import numpy as np

%matplotlib inline
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('files/imdb-movies.csv')
df

id    imdb_id  popularity     budget     revenue  \
0      135397  tt0369610   32.985763  150000000  1513528810   
1       76341  tt1392190   28.419936  150000000   378436354   
2      262500  tt2908446   13.112507  110000000   295238201   
3      140607  tt2488496   11.173104  200000000  2068178225   
4      168259  tt2820852    9.335014  190000000  1506249360   
5      281957  tt1663202    9.110700  135000000   532950503   
6       87101  tt1340138    8.654359  155000000   440603537   
7      286217  tt3659388    7.667400  108000000   595380321   
8      211672  tt2293640    7.404165   74000000  1156730962   
9      150540  tt2096673    6.326804  175000000   853708609   
10     206647  tt2379713    6.200282  245000000   880674609   
11      76757  tt1617661    6.189369  176000003   183987723   
12     264660  tt0470752    6.118847   15000000    36869414   
13     257344  tt2120120    5.984995   88000000   243637091   
14      99861  tt2395427    5.944927  280000000  1405035767   
15     273248  tt3460252    5.898400   44000000   155760117   
16     260346  tt2446042    5.749758   48000000   325771424   
17     102899  tt0478970    5.573184  130000000   518602163   
18     150689  tt1661199    5.556818   95000000   542351353   
19     131634  tt1951266    5.476958  160000000   650523427   
20     158852  tt1964418    5.462138  190000000   209035668   
21     307081  tt1798684    5.337064   30000000    91709827   
22     254128  tt2126355    4.907832  110000000   470490832   
23     216015  tt2322441    4.710402   40000000   569651467   
24     318846  tt1596363    4.648046   28000000   133346506   
25     177677  tt2381249    4.566713  150000000   682330139   
26     214756  tt2637276    4.564549   68000000   215863606   
27     207703  tt2802144    4.503789   81000000   403802136   
28     314365  tt1895587    4.062293   20000000    88346473   
29     294254  tt4046784    3.968891   61000000   311256926   
...       ...        ...         ...        ...         ...   
10836   38720  tt0061170    0.239435          0           0   
10837   19728  tt0060177    0.291704          0           0   
10838   22383  tt0060862    0.151845          0           0   
10839   13353  tt0060550    0.276133          0           0   
10840   34388  tt0060437    0.102530          0           0   
10841   42701  tt0062262    0.264925      75000           0   
10842   36540  tt0061199    0.253437          0           0   
10843   29710  tt0060588    0.252399          0           0   
10844   23728  tt0059557    0.236098          0           0   
10845    5065  tt0059014    0.230873          0           0   
10846   17102  tt0059127    0.212716          0           0   
10847   28763  tt0060548    0.034555          0           0   
10848    2161  tt0060397    0.207257    5115000    12000000   
10849   28270  tt0060445    0.206537          0           0   
10850   26268  tt0060490    0.202473          0           0   
10851   15347  tt0060182    0.342791          0           0   
10852   37301  tt0060165    0.227220          0           0   
10853   15598  tt0060086    0.163592          0           0   
10854   31602  tt0060232    0.146402          0           0   
10855   13343  tt0059221    0.141026     700000           0   
10856   20277  tt0061135    0.140934          0           0   
10857    5921  tt0060748    0.131378          0           0   
10858   31918  tt0060921    0.317824          0           0   
10859   20620  tt0060955    0.089072          0           0   
10860    5060  tt0060214    0.087034          0           0   
10861      21  tt0060371    0.080598          0           0   
10862   20379  tt0060472    0.065543          0           0   
10863   39768  tt0060161    0.065141          0           0   
10864   21449  tt0061177    0.064317          0           0   
10865   22293  tt0060666    0.035919      19000           0   

                                         original_title  \
0                                        Jura

### Drop columns without neccesary information and remove all records with no financial information -- Pay close attention to things that don't tell you anything regarding financial data

In [3]:
df.drop(df.query('budget == 0 or revenue == 0').index, inplace = True)
df.drop(['id','popularity','imdb_id','homepage','budget', 'runtime','tagline','keywords','vote_count','vote_average','release_date','budget_adj','overview'], axis=1, inplace=True)

In [4]:
df.dropna(inplace = True)
df.reset_index(drop = True, inplace = True)
df

revenue                                     original_title  \
0     1513528810                                     Jurassic World   
1      378436354                                 Mad Max: Fury Road   
2      295238201                                          Insurgent   
3     2068178225                       Star Wars: The Force Awakens   
4     1506249360                                          Furious 7   
5      532950503                                       The Revenant   
6      440603537                                 Terminator Genisys   
7      595380321                                        The Martian   
8     1156730962                                            Minions   
9      853708609                                         Inside Out   
10     880674609                                            Spectre   
11     183987723                                  Jupiter Ascending   
12      36869414                                         Ex Machina   
13     243637091                                             Pixels   
14    1405035767                            Avengers: Age of Ultron   
15     155760117                                  The Hateful Eight   
16     325771424                                            Taken 3   
17     518602163                                            Ant-Man   
18     542351353                                         Cinderella   
19     650523427              The Hunger Games: Mockingjay - Part 2   
20     209035668                                       Tomorrowland   
21      91709827                                           Southpaw   
22     470490832                                        San Andreas   
23     569651467                               Fifty Shades of Grey   
24     133346506                                      The Big Short   
25     682330139                 Mission: Impossible - Rogue Nation   
26     215863606                                              Ted 2   
27     403802136                       Kingsman: The Secret Service   
28      88346473                                          Spotlight   
29     311256926                     Maze Runner: The Scorch Trials   
...          ...                                                ...   
3776   163214286                                 The Sound of Music   
3777   111721910                                     Doctor Zhivago   
3778    29950000  Those Magnificent Men in Their Flying Machines...   
3779    12000000                       The Greatest Story Ever Told   
3780    81974493                    On Her Majesty's Secret Service   
3781   102308889                 Butch Cassidy and the Sundance Kid   
3782    44785053                                    Midnight Cowboy   
3783      638641                                     The Wild Bunch   
3784   181813770                                             Grease   
3785   187884007                                             Jaws 2   
3786    55000000                                   Dawn of the Dead   
3787   300218018                                           Superman   
3788    70000000                                          Halloween   
3789   141000000                                       Animal House   
3790    50000000                                    The Deer Hunter   
3791    35000000                                   Midnight Express   
3792    30471420                              The Lord of the Rings   
3793    14560084                                  Death on the Nile   
3794          11                                           F.I.S.T.   
3795     7230000                             Force 10 from Navarone   
3796    22765081                                             Convoy   
3797    24046533                     Invasion of the Body Snatchers   
3798    21049053                                            The Wiz   
3799    26518355                                    Damien: Omen II   
3800     3713768                 

In [5]:
production_records = df[df.production_companies.str.contains('|')] 
production_records.production_companies = production_records.production_companies.apply(lambda x: x.split('|'))

df_pc = pd.DataFrame(columns = df.columns)
for i in range(len(production_records)):
    record = production_records.iloc[i] # returns all data for every single record with the ability to access columns
    for production_company in production_records.production_companies[i]: # loop through ALL production_companies from EACH record
        df_pc = df_pc.append(pd.DataFrame([[record.revenue, record.original_title, record.cast, record.director,record.genres, production_company, record.release_year, record.revenue_adj]], columns = df.columns))
            
df_pc.drop('genres', axis=1, inplace=True)
df_pc.reset_index(drop=True,inplace=True)



In [6]:
df_pc

revenue                   original_title  \
0      1513528810                   Jurassic World   
1      1513528810                   Jurassic World   
2      1513528810                   Jurassic World   
3      1513528810                   Jurassic World   
4      1513528810                   Jurassic World   
5       378436354               Mad Max: Fury Road   
6       378436354               Mad Max: Fury Road   
7       295238201                        Insurgent   
8       295238201                        Insurgent   
9       295238201                        Insurgent   
10      295238201                        Insurgent   
11     2068178225     Star Wars: The Force Awakens   
12     2068178225     Star Wars: The Force Awakens   
13     2068178225     Star Wars: The Force Awakens   
14     1506249360                        Furious 7   
15     1506249360                        Furious 7   
16     1506249360                        Furious 7   
17     1506249360                        Furious 7   
18     1506249360                        Furious 7   
19      532950503                     The Revenant   
20      532950503                     The Revenant   
21      532950503                     The Revenant   
22      532950503                     The Revenant   
23      532950503                     The Revenant   
24      440603537               Terminator Genisys   
25      440603537               Terminator Genisys   
26      595380321                      The Martian   
27      595380321                      The Martian   
28      595380321                      The Martian   
29      595380321                      The Martian   
...           ...                              ...   
10553    70000000                        Halloween   
10554   141000000                     Animal House   
10555   141000000                     Animal House   
10556   141000000                     Animal House   
10557    50000000                  The Deer Hunter   
10558    50000000                  The Deer Hunter   
10559    35000000                 Midnight Express   
10560    35000000                 Midnight Express   
10561    30471420            The Lord of the Rings   
10562    14560084                Death on the Nile   
10563    14560084                Death on the Nile   
10564          11                         F.I.S.T.   
10565     7230000           Force 10 from Navarone   
10566    22765081                           Convoy   
10567    24046533   Invasion of the Body Snatchers   
10568    24046533   Invasion of the Body Snatchers   
10569    21049053                          The Wiz   
10570    21049053                          The Wiz   
10571    26518355                  Damien: Omen II   
10572    26518355                  Damien: Omen II   
10573     3713768                   Watership Down   
10574     3713768                   Watership Down   
10575    33736689  Who's Afraid of Virginia Woolf?   
10576    13000000                     Torn Curtain   
10577     6000000                        El Dorado   
10578     6000000                        El Dorado   
10579    20000000                 The Sand Pebbles   
10580    20000000                 The Sand Pebbles   
10581    20000000                 The Sand Pebbles   
10582    12000000                 Fantastic Voyage   

                                                    cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
2      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
3      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
4      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
5      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
6      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
7      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
8      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
9      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
1

In [7]:
production_records = df[df.genres.str.contains('|')] 
production_records.genres = production_records.genres.apply(lambda x: x.split('|'))

df_g = pd.DataFrame(columns = df.columns)
for i in range(len(production_records)):
    record = production_records.iloc[i] # returns all data for every single record with the ability to access columns
    for genre in production_records.genres[i]: # loop through ALL production_companies from EACH record
        df_g = df_g.append(pd.DataFrame([[record.revenue, record.original_title, record.cast, record.director,genre, record.production_companies, record.release_year, record.revenue_adj]], columns = df.columns))
            
# df_g.drop('production_companies', axis=1, inplace=True)
df_g.reset_index(drop=True,inplace=True)



In [8]:
df_g.head()

revenue      original_title  \
0  1513528810      Jurassic World   
1  1513528810      Jurassic World   
2  1513528810      Jurassic World   
3  1513528810      Jurassic World   
4   378436354  Mad Max: Fury Road   

                                                cast         director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
1  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
2  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
3  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...  Colin Trevorrow   
4  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...    George Miller   

            genres                               production_companies  \
0           Action  Universal Studios|Amblin Entertainment|Legenda...   
1        Adventure  Universal Studios|Amblin Entertainment|Legenda...   
2  Science Fiction  Universal Studios|Amblin Entertainment|Legenda...   
3         Thriller  Universal Studios|Amblin Entertainment|Legenda...   
4           Action  Village Roadshow Pictures|Kennedy Miller Produ...   

  release_year   revenue_adj  
0         2015  1.392446e+09  
1         2015  1.392446e+09  
2         2015  1.392446e+09  
3         2015  1.392446e+09  
4         2015  3.481613e+08

### Data Cleaning

In [180]:
# Delete all records with null, or empty values



#### Here's a helpful hint from my own analysis when I ran this the first time. This may help shed light on what your data set should look like.

#### If I created one record for each the `production_companies` a movie was release under and one record each for `genres`<br>and tried to run calculations, it wouldn't work because for many records, the amount of `production_companies`<br>and `genres` aren't the same, so I'll create 2 dataframes; one w/o a `production_companies` column and one w/o a `genres` columns

In [9]:
# df_genres = df.drop('production_companies', axis=1, inplace=False)
# df_genres
# df_production_companies = df.drop('genres', axis=1, inplace=False)
# df_production_companies
df_g


revenue                   original_title  \
0      1513528810                   Jurassic World   
1      1513528810                   Jurassic World   
2      1513528810                   Jurassic World   
3      1513528810                   Jurassic World   
4       378436354               Mad Max: Fury Road   
5       378436354               Mad Max: Fury Road   
6       378436354               Mad Max: Fury Road   
7       378436354               Mad Max: Fury Road   
8       295238201                        Insurgent   
9       295238201                        Insurgent   
10      295238201                        Insurgent   
11     2068178225     Star Wars: The Force Awakens   
12     2068178225     Star Wars: The Force Awakens   
13     2068178225     Star Wars: The Force Awakens   
14     2068178225     Star Wars: The Force Awakens   
15     1506249360                        Furious 7   
16     1506249360                        Furious 7   
17     1506249360                        Furious 7   
18      532950503                     The Revenant   
19      532950503                     The Revenant   
20      532950503                     The Revenant   
21      532950503                     The Revenant   
22      440603537               Terminator Genisys   
23      440603537               Terminator Genisys   
24      440603537               Terminator Genisys   
25      440603537               Terminator Genisys   
26      595380321                      The Martian   
27      595380321                      The Martian   
28      595380321                      The Martian   
29     1156730962                          Minions   
...           ...                              ...   
10155    22765081                           Convoy   
10156    22765081                           Convoy   
10157    24046533   Invasion of the Body Snatchers   
10158    24046533   Invasion of the Body Snatchers   
10159    24046533   Invasion of the Body Snatchers   
10160    24046533   Invasion of the Body Snatchers   
10161    21049053                          The Wiz   
10162    21049053                          The Wiz   
10163    21049053                          The Wiz   
10164    21049053                          The Wiz   
10165    21049053                          The Wiz   
10166    26518355                  Damien: Omen II   
10167    26518355                  Damien: Omen II   
10168    26518355                  Damien: Omen II   
10169    26518355                  Damien: Omen II   
10170     3713768                   Watership Down   
10171     3713768                   Watership Down   
10172     3713768                   Watership Down   
10173    33736689  Who's Afraid of Virginia Woolf?   
10174    13000000                     Torn Curtain   
10175    13000000                     Torn Curtain   
10176     6000000                        El Dorado   
10177     6000000                        El Dorado   
10178    20000000                 The Sand Pebbles   
10179    20000000                 The Sand Pebbles   
10180    20000000                 The Sand Pebbles   
10181    20000000                 The Sand Pebbles   
10182    20000000                 The Sand Pebbles   
10183    12000000                 Fantastic Voyage   
10184    12000000                 Fantastic Voyage   

                                                    cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
2      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
3      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
4      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
5      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
6      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
7      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
8      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
9      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
1

<a id="eda"></a>
## Exploratory Data Analysis

> Use Matplotlib to display your data analysis

### Which production companies released the most movies in the last 10 years? Display the top 5 production companies.

In [12]:
last_10 = df_pc.query('release_year > 2007').production_companies.value_counts().nlargest(10)
last_10

Relativity Media                          90
Universal Pictures                        89
Warner Bros.                              87
Columbia Pictures                         73
Paramount Pictures                        60
Twentieth Century Fox Film Corporation    52
New Line Cinema                           45
Walt Disney Pictures                      41
Summit Entertainment                      38
Lionsgate                                 37
Name: production_companies, dtype: int64

### What 5 movie genres grossed the highest all-time?

In [16]:
high_gross = df_g.groupby('genres')['revenue'].sum().nlargest(5)
high_gross

genres
Action       169790679429
Adventure    163439947064
Comedy       132020867028
Drama        130323128041
Thriller     117587434010
Name: revenue, dtype: int64

### Who are the top 5 grossing directors?

In [14]:
# high_gross = df.groupby('director', as_index=False).mean().sort_values(['director'], ascending=True)
high_director = df.groupby('director')['revenue'].sum().nlargest(5)
high_director

director
Steven Spielberg     9018563772
Peter Jackson        6523244659
James Cameron        5841894863
Michael Bay          4917208171
Christopher Nolan    4167548502
Name: revenue, dtype: int64

### Compare the revenue of the highest grossing movies of all time.

In [15]:
highest_movie = df.groupby('original_title')['revenue'].sum().nlargest(5)
highest_movie

original_title
Avatar                          2781505847
Star Wars: The Force Awakens    2068178225
Titanic                         1845034188
The Avengers                    1568080742
Jurassic World                  1513528810
Name: revenue, dtype: int64

<a id="conclusions"></a>
## Conclusions

> Using the cell below, write a brief conclusion of what you have found from the anaylsis of the data. The Cell below will allow you to write plan text instead of code.

FINDINGS:
    =>Relativity Media Production Company released 90 movies in the last 10 years and that made them
      1st in the Production Company.
    =>Action movie with 169790679429 movie genres occupied the first position in the genres category.
    =>Highest Director Gross was Steven Spielberg with 9018563772 total.
    =>The revenue of the highest grossing movies of all time was Avatar with 2781505847.
    